datafusion/datasource-parquet/src/row_filter.rs

```diff
@@ -537,12 +464,20 @@ pub fn build_row_filter(
     // `a = 1 AND b = 2 AND c = 3` -> [`a = 1`, `b = 2`, `c = 3`]
     let predicates = split_conjunction(expr);
 
+    let file_schema = Arc::new(file_schema.clone());
+    let table_schema = Arc::new(table_schema.clone());
```

Comment:
We could change the signature of `build_row_filter` since the caller might have an `Arc`'d version already, but since it's `pub` that would introduce more breaking changes and the clone seemed cheap enough. Open to doing that though.

Reply:
I think you can avoid cloning the schema with a pretty simple change. Here is a proposal:
- https://github.com/pydantic/datafusion/pull/9